tests/ts_simple_select_is_null.erl

```diff
@@ -0,0 +1,134 @@
+%% NULL predicates proves that each predicate is correctly implemented.
+
+-module(ts_simple_select_is_null).
+
+-behavior(riak_test).
+
+-export([confirm/0]).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-define(SPANNING_STEP, (1000)).
+-define(TEST_TYPE, normal).
+
+confirm() ->
+    %% will fail if ts_simple_put_all_null_datatypes fails
+    DDL =
+        "CREATE TABLE GeoCheckin ("
+        " myfamily    varchar     not null,"
+        " myseries    varchar     not null,"
+        " time        timestamp   not null,"
+        " myvarchar   varchar             ,"
+        " myint       sint64              ,"
+        " myfloat     double              ,"
+        " mybool      boolean             ,"
+        " mytimestamp timestamp           ,"
+        " PRIMARY KEY ((myfamily, myseries, quantum(time, 15, 'm')),"
+        " myfamily, myseries, time))",
+    Family = <<"family1">>,
+    Series = <<"seriesX">>,
+    N = 10,
+    Data = make_data(N, Family, Series, []),
+    ClusterConn = ts_util:cluster_and_connect(single),
+    Got = ts_util:ts_put(ClusterConn, ?TEST_TYPE, DDL, Data),
+    ?assertEqual(ok, Got),
+    Qry = "SELECT * FROM GeoCheckin WHERE"
+          " myfamily='" ++ binary_to_list(Family) ++ "'"
+          " AND myseries='" ++ binary_to_list(Series) ++ "'"
+          " AND time >= 1000 AND time <= " ++ integer_to_list(N * 1000) ++
+          " AND myvarchar IS NULL",
+    {ok, {_Fields, Rows}} = ts_util:ts_query(ClusterConn, ?TEST_TYPE, DDL, Data, Qry),
+    ?assertNotEqual(0, length(Rows)),
+    NullableFields = [ "myvarchar", "myint", "myfloat", "mybool", "mytimestamp" ],
+    lists:foreach(fun (Field) ->
+                query_field(Field, ClusterConn, DDL, Data, Family, Series, N)
+        end, NullableFields),
+    pass.
+
+query_field(Field, ClusterConn, DDL, Data, Family, Series, N) ->
+    RowsAll = query_all(ClusterConn, DDL, Data, Family, Series, N),
+    RowsIsNull = query_is_null(ClusterConn, DDL, Data, Family, Series, N, Field),
+    RowsIsNotNull = query_is_not_null(ClusterConn, DDL, Data, Family, Series, N, Field),
+    ?assertEqual(RowsAll, RowsIsNull + RowsIsNotNull).
+
+query_base(Family, Series, N) ->
+    "SELECT * FROM GeoCheckin WHERE"
+    " myfamily='" ++ binary_to_list(Family) ++ "'"
+    " AND myseries='" ++ binary_to_list(Series) ++ "'"
+    " AND time >= 1000 AND time <= " ++ integer_to_list(N * 1000).
+
+query_all(ClusterConn, DDL, Data, Family, Series, N) ->
+    Qry = query_base(Family, Series, N),
+    {ok, {_Fields, Rows}} = ts_util:ts_query(ClusterConn, ?TEST_TYPE, DDL, Data, Qry),
+    RowsN = length(Rows),
+    ?assertNotEqual(0, RowsN),
+    RowsN.
+
+query_is_null(ClusterConn, DDL, Data, Family, Series, N, Field) ->
+    Qry = query_base(Family, Series, N) ++
+          " AND " ++ Field ++ " IS NULL",
+    {ok, {_Fields, Rows}} = ts_util:ts_query(ClusterConn, ?TEST_TYPE, DDL, Data, Qry),
+    RowsN = length(Rows),
+    ?assertNotEqual(0, RowsN),
```

Comment:
There is a comment at the head of the test module that explains why the need to check for exact set sizes for each subset is overkill, but I'm making the change regardless. It is better if we all can understand the code.